build.xml

```diff
@@ -33,6 +27,21 @@
 	<property name="dwr-maven-jdk14-jar" value="${dist}/lib/${artifactId}-${version}-jdk14.jar" />
 	<!-- End Properties for Maven -->
 
+	<!-- Get build number from Github CI -->
+	<property environment="envgithub"/>
+
+	<!-- Updates version-properties with a build number if this is a CI run -->
+	<target name="updateVersionPropertiesWithBuildNumber">
+		<if>
+			<isset property="envgithub.GITHUB_RUN_ID" />
+			<then>
+				<propertyfile file="${target.root}/classes/main/dwr-version.properties">
+					<entry key="build.number" value="${envgithub.GITHUB_RUN_ID}" />
+				</propertyfile>
+			</then>
+		</if>
+	</target>
+
```

Comment:
Ok so you are now back at roughly the same solution as I initially commented on. This is not a big deal so do like you want, but the reason I preferred that you do:
  build.number = envgithub.GITHUB_RUN_ID
in the property section and then use build.number, is to isolate as much as possible of build.xml from external system dependencies. In theory we could get a build number from any place and then we would just edit the property assignment. Using envgithub.GITHUB_RUN_ID or any other envgithub.<var> inside target code is less separated IMO even if those targets are pushed to top of the file.